gremlin-go/driver/client.go

```diff
@@ -41,7 +41,7 @@ type Client struct {
 }
 
 // NewClient creates a Client and configures it with the given parameters.
-func NewClient(host string, port int, configurations ...func(settings *ClientSettings)) *Client {
+func NewClient(host string, port int, configurations ...func(settings *ClientSettings)) (*Client, error) {
```

Comment:
This _looks_ like the functional options pattern but all the settings are exported which isn't usually the case with that pattern. Do the settings need to be exported?